### PR TITLE
Refine modeling intro

### LINKED ISSUE #43: more astronomy focused Modeling introduction?
The modelling capability is new for me, so I read it with interest. Here a suggestion to make it even more interesting to readers. (Feel free to ignore though, the section is already nice.)

Perhaps it can be elaborated on why the modelling is added to astropy at all? As in, what are the astronomical drivers to have modelling in an astronomy package? Why did we not use/wrap/wrote a general modelling package instead?

From the rest of the section I can guess, e.g. that a varying-over-the-FoV-PSF-model is a typical astronomy thing, or that we want to use Quantity objects, etc. Perhaps we can put the astronomy-specific 'why' at the start of the Modeling section?

(Also, it isn't really necessary to have a subsubsection (3.7.1) called 'Overview'. That text can just be in subsection 3.7 directly I think.)

### PR DESCRIPTION
**Pull request by pllim**
Fix #43 

Please review -- @hugobuddel and @nden. I took the liberties to edit the text given in https://github.com/astropy/astropy-v2.0-paper/issues/43#issuecomment-349996742 a little, hope you don't mind.

### CODE REVIEW

**Comment by nden** [comment]
@pllim Thanks for submitting a PR.

**Comment by hugobuddel** [comment]
Thanks, @nden and @pllim, the section is nicer to read now IMHO.

Reading the source, I see that there already used to be a sentence that adding unit support makes the modeling packages especially useful for astronomy, which was commented out. (I just read the pdf earlier.) The new introduction has a similar sentence again.

File: main.tex [hunk 1]
```diff
@@ -883,15 +883,20 @@ \subsection{Modeling}
 % Nadia, Lim
 % Tom R. -- unit support
 
-The \package{astropy.modeling} subpackage provides a framework for representing
-analytical models and performing model evaluation and parameter fitting. Models and
-fitters are independent of each other: a model can be fit with different
+Tha \package{astropy.modeling} subpackage provides a framework for representing
```
Comment (adrn) [inline main.tex]: "Tha" -> "The"

File: main.tex [hunk 2]
```diff
@@ -883,15 +883,20 @@ \subsection{Modeling}
...
+analytical models and performing model evaluation and parameter fitting.
+The main motivation for it was creating a framework
```
Comment (adrn) [inline main.tex]: A few grammar things: "The main motivation for this functionality was to create a framework that allows arbitrary combination of models to..."

File: main.tex [hunk 3]
```diff
@@ -883,15 +883,20 @@ \subsection{Modeling}
...
+which allows arbitrary combination of models to support the Generalized World
+Coordinate System (GWCS) package.\footnote{\url{https://github.com/spacetelescope/gwcs}}
+The current FITS WCS model lacks the flexibility to represent
```
Comment (adrn) [inline main.tex]: The use of "model" here is different than in above (one of the reasons I dislike the word - it is too overloaded in english!). Maybe change to "The current FITS WCS specification lacks the..."

File: main.tex [hunk 4]
```diff
@@ -883,15 +883,20 @@ \subsection{Modeling}
...
+arbitrary distortions and does not meet the needs of many types of current instrumentation.
+The addition of units to modeling made it attractive for representing
```
Comment (adrn) [inline main.tex]: I don't like the passive voice here. Maybe: "The fact that the \astropypkg modeling framework now supports propagating units also makes it a useful tool for representing and fitting astrophysical models within data analysis tools."

**Review by adrn — CHANGES_REQUESTED** [review]

**Comment by pllim** [comment]
Thank you for the reviews. I think I have addressed all the comments.

**Review by adrn — APPROVED** [review]

**Pull request merged**